images/image.go

```diff
@@ -667,7 +691,17 @@ func cryptChildren(ctx context.Context, cs content.Store, desc ocispec.Descripto
 		}
 
 		ref := fmt.Sprintf("manifest-%s", newDesc.Digest.String())
-		if err := content.WriteBlob(ctx, cs, ref, bytes.NewReader(mb), newDesc, content.WithLabels(labels), content.WithExpiration(time.Minute*5)); err != nil {
+		if rsrcFn != nil {
+			r := leases.Resource{
+				ID:   newDesc.Digest.String(),
+				Type: "content",
+			}
+			if err := rsrcFn(r); err != nil {
+				return ocispec.Descriptor{}, false, errors.Wrap(err, "Unable to add resource to lease")
+			}
+		}
+
+		if err := content.WriteBlob(ctx, cs, ref, bytes.NewReader(mb), newDesc, content.WithLabels(labels)); err != nil {
```

Comment:
This would look better I think:
```
err := content.WriteBlobLeased(ctx, cs, ref, bytes.NewReader(mb), newDesc, rsrcFn, content.WithLabels(labels))
```